### ISSUE TITLE
Sniff `SlevomatCodingStandard.PHP.UselessParentheses` with option enableCheckAroundNew should consider PHP version

### ISSUE THREAD

**Issue by stlrnz** [slevomat/coding-standard#1805]
In https://github.com/slevomat/coding-standard/pull/1789 you added a great new option `enableCheckAroundNew ` to the sniff `SlevomatCodingStandard.PHP.UselessParentheses` to check for useless parentheses around the `new` operator.

However, this feature is only available since PHP 8.4: https://wiki.php.net/rfc/new_without_parentheses
The sniff reports an error regardless of the used PHP version. This is an issue for coding standards supporting older versions of PHP. It effectively makes this feature unusable in this/my scenario.

Could you update the sniff and make the flag `enableCheckAroundNew` aware of the PHP version?

**Comment by kukulich** [comment]
It would be BC break.

**Comment by roslov** [comment]
It seems the fix does not work correctly.

<img width="1071" height="1136" alt="Image" src="https://github.com/user-attachments/assets/be140373-3172-47a1-bd46-9523bb60ee10" />


It detects my PHP 8.3 version incorrectly. Therefore, PHPCS returns error `[x] Useless parentheses. (SlevomatCodingStandard.PHP.UselessParentheses.UselessParentheses)`.

Probably, `getPhpVersion()` should be changed from:
```php
public static function getPhpVersion(): ?int
{
	$phpVersion = Config::getConfigData('php_version');

	return $phpVersion !== null ? (int) $phpVersion : null;
}
```
to
```php
public static function getPhpVersion(): ?int
{
	$phpVersion = Config::getConfigData('php_version') ?? PHP_VERSION_ID;

	return $phpVersion !== null ? (int) $phpVersion : null;
}
```

**Comment by roslov** [comment]
Then probably this fix should help:

<img width="1532" height="1067" alt="Image" src="https://github.com/user-attachments/assets/3aeddabc-1de9-400c-a24d-752fe8b347dc" />

**Comment by kukulich** [comment]
As I said before - the behaviour is expected and intentional (to not create BC break). You can set your required PHP version in your PHPCS configuration.

**Comment by kukulich** [comment]
Yes, it's 100 % better. PHPCS can run on any PHP version, but your configuration can specify what is the target PHP version.
